modules/charon-core/src/main/java/org/wso2/charon3/core/config/SCIMSystemSchemaExtensionBuilder.java

```diff
@@ -0,0 +1,163 @@
+
+        extensionRootAttributeURI = SYSTEM_USER_SCHEMA_URI;
+    }
+
+    /**
+     * Get the extension schema.
+     *
+     * @return The extension schema.
+     */
+    public AttributeSchema getExtensionSchema() {
+
+        return extensionSchema;
+    }
+
+    /**
+     * Build the system schema extension from the config file.
+     *
+     * @param configFilePath            Path to the config file.
+     * @throws CharonException          If an error occurred while reading the config file.
+     * @throws InternalErrorException   If an error occurred while building the schema.
+     */
+    public void buildSystemSchemaExtension(String configFilePath) throws CharonException, InternalErrorException {
+
+        File provisioningConfig = new File(configFilePath);
+        try (InputStream configFilePathInputStream = new FileInputStream(provisioningConfig)) {
+            buildSystemSchemaExtension(configFilePathInputStream);
+        } catch (FileNotFoundException e) {
+            throw new CharonException(configFilePath + " file not found!", e);
+        } catch (JSONException e) {
+            throw new CharonException("Error while parsing " + configFilePath + " file!", e);
+        } catch (IOException e) {
+            throw new CharonException("Error while closing " + configFilePath + " file!", e);
+        }
+    }
+
+    /**
+     * Build the system schema extension from the input stream.
+     *
+     * @param inputStream               The input stream.
+     * @throws CharonException          If an error occurred while reading the configuration.
+     * @throws InternalErrorException   If an error occurred while building the schema.
+     */
+    public void buildSystemSchemaExtension(InputStream inputStream) throws CharonException, InternalErrorException {
+
+        readConfiguration(inputStream);
+        for (Map.Entry<String, ExtensionAttributeSchemaConfig> attributeSchemaConfig : extensionConfig.entrySet()) {
+            // If there are no children it is a simple attribute, build it.
+            if (!attributeSchemaConfig.getValue().hasChildren()) {
+                buildSimpleAttributeSchema(attributeSchemaConfig.getValue(), attributeSchemas);
+            } else {
+                // Need to build child schemas first.
+                buildComplexAttributeSchema(attributeSchemaConfig.getValue(), attributeSchemas, extensionConfig);
+            }
+        }
+
+        extensionSchema = attributeSchemas.get(extensionRootAttributeURI);
+    }
+
+    /**
+     * Read the configuration from the input stream.
+     *
+     * @param inputStream               The input stream.
+     * @throws CharonException          If an error occurred while reading the configuration.
+     */
+    public void readConfiguration(InputStream inputStream) throws CharonException {
+
+        Scanner scanner = new Scanner(inputStream, "utf-8").useDelimiter("\\A");
+        String jsonString = scanner.hasNext() ? scanner.next() : "";
```

Comment:
Let's check whether we have constants for these strings. May be from common string libraries, or defined constants in this repo.

Reply:
Used string literals for UTF-8 and the empty string. Extracted the delimiter to a class level constant.
Ref: dab02913610656de861ffb7a6360ae268ae16a95